app/renderer/lib/client-frameworks/ruby.js

```diff
@@ -135,36 +163,79 @@ driver.quit`;
     return `driver.set_clipboard content: '${clipboardText}'`;
   }
 
-  codeFor_pressKeyCode (varNameIgnore, varIndexIgnore, keyCode, metaState, flags) {
-    return `driver.press_keycode ${keyCode}, ${metaState}, ${flags}`;
+  // File Transfer
+
+  codeFor_pushFile (varNameIgnore, varIndexIgnore, pathToInstallTo, fileContentString) {
+    return `driver.push_file '${pathToInstallTo}', '${fileContentString}'`;
   }
 
-  codeFor_longPressKeyCode (varNameIgnore, varIndexIgnore, keyCode, metaState, flags) {
-    return `driver.long_press_keycode ${keyCode}, ${metaState}, ${flags}`;
+  codeFor_pullFile (varNameIgnore, varIndexIgnore, pathToPullFrom) {
+    return `driver.pull_file '${pathToPullFrom}'`;
   }
 
-  codeFor_hideKeyboard () {
-    return `driver.hide_keyboard`;
+  codeFor_pullFolder (varNameIgnore, varIndexIgnore, folderToPullFrom) {
+    return `driver.pull_folder '${folderToPullFrom}'`;
   }
 
-  codeFor_isKeyboardShown () {
-    return `is_keyboard_shown = driver.is_keyboard_shown`;
+  // Device Interaction
+
+  codeFor_shake () {
+    return this.codeFor_executeScriptNoArgs('mobile: shake');
   }
 
-  codeFor_pushFile (varNameIgnore, varIndexIgnore, pathToInstallTo, fileContentString) {
-    return `driver.push_file '${pathToInstallTo}', '${fileContentString}'`;
+  codeFor_lock (varNameIgnore, varIndexIgnore, seconds) {
+    return this.codeFor_executeScriptWithArgs('mobile: lock', [{seconds}]);
   }
 
-  codeFor_pullFile (varNameIgnore, varIndexIgnore, pathToPullFrom) {
-    return `driver.pull_file '${pathToPullFrom}'`;
+  codeFor_unlock () {
+    // TODO: UiAutomator2 requires arguments, XCUITest does not
+    return this.codeFor_executeScriptNoArgs('mobile: unlock');
   }
 
-  codeFor_pullFolder (varNameIgnore, varIndexIgnore, folderToPullFrom) {
-    return `driver.pull_folder '${folderToPullFrom}'`;
+  codeFor_isLocked () {
+    return `is_locked = ${this.codeFor_executeScriptNoArgs('mobile: isLocked')}`;
+  }
+
+  codeFor_rotateDevice () {
+    return `# Not supported: rotateDevice`;
```

Comment:
Not for this issue itself. but maybe we no longer function this method itself.
https://github.com/search?q=org%3Aappium%20mobileRotation&type=code

Reply:
-> https://github.com/appium/appium/issues/19378

Reply:
Yes, they should likely be replaced with `getRotation`/`setRotation`. Not sure if the current drivers implement it though.